src/mame/drivers/gradius3.cpp

```diff
@@ -165,23 +122,27 @@ void gradius3_state::gradius3_map(address_map &map)
 	map(0x0e0000, 0x0e0001).w("watchdog", FUNC(watchdog_timer_device::reset16_w));
 	map(0x0e8000, 0x0e8000).w("soundlatch", FUNC(generic_latch_8_device::write));
 	map(0x0f0000, 0x0f0001).w(this, FUNC(gradius3_state::sound_irq_w));
-	map(0x100000, 0x103fff).ram().share("share1");
-	map(0x14c000, 0x153fff).rw(this, FUNC(gradius3_state::k052109_halfword_r), FUNC(gradius3_state::k052109_halfword_w));
-	map(0x180000, 0x19ffff).ram().w(this, FUNC(gradius3_state::gradius3_gfxram_w)).share("k052109");
+	map(0x100000, 0x103fff).ram().share("shared_ram");
+	map(0x14c000, 0x153fff).rw(m_k052109, FUNC(k052109_device::read), FUNC(k052109_device::write)).umask(0x00ff);
+	/* is this a bug in the game or something else? */
```

Comment:
What you've come across here is known as "byte smearing." Basically, when the MC68000 writes data to the chip with move.b or some other instruction, it will be present on both halves of the data bus. It's up to the address decoding circuit to only select writes directed to one half, but it's often easier for hardware to just ignore the extra control signals and allow writes to either byte to affect the peripheral.

I've implemented a standard way for specifying byte-smeared accesses in 16-bit address maps: just add `.cswidth(16)` to the entry. See cyclwarr in tatsumi.cpp for an example of this.